sdk/cosmos/azure_data_cosmos/src/retry_policies/mod.rs

```diff
@@ -0,0 +1,319 @@
+///
+/// # Fields
+/// * `should_retry` - Whether the operation should be retried
+/// * `backoff_time` - Duration to wait before retrying (meaningful only if `should_retry` is true)
+#[derive(Debug, Clone)]
+pub struct ShouldRetryResult {
+    pub should_retry: bool,
+    pub back_off_time: Duration,
+}
+
+impl ShouldRetryResult {
+    /// Creates a result indicating the operation should not be retried
+    ///
+    /// # Example
+    /// ```
+    /// use azure_data_cosmos::retry_policies::ShouldRetryResult;
+    ///
+    /// let result = ShouldRetryResult::no_retry();
+    /// assert!(!result.should_retry);
+    /// ```
+    pub fn no_retry() -> Self {
+        Self {
+            should_retry: false,
+            back_off_time: Duration::ZERO,
+        }
+    }
+
+    /// Creates a result indicating the operation should be retried after a delay
+    ///
+    /// # Arguments
+    /// * `backoff` - The duration to wait before retrying
+    ///
+    /// # Example
+    /// ```
+    /// use azure_data_cosmos::retry_policies::ShouldRetryResult;
+    /// use std::time::Duration;
+    ///
+    /// let result = ShouldRetryResult::retry_after(Duration::from_secs(5));
+    /// assert!(result.should_retry);
+    /// assert_eq!(result.back_off_time, Duration::from_secs(5));
+    /// ```
+    pub fn retry_after(backoff: Duration) -> Self {
+        Self {
+            should_retry: true,
+            back_off_time: backoff,
+        }
+    }
+}
+
+/// Trait defining the retry policy interface for Cosmos DB operations
+///
+/// This trait provides a contract for implementing retry logic for transient failures
+/// in Azure Cosmos DB operations. Implementers can define custom retry behavior for
+/// both exceptions (errors) and HTTP responses based on their specific requirements.
+#[async_trait]
+pub trait RetryPolicy: Send + Sync {
+    /// Called before sending a request to allow policy-specific modifications
+    ///
+    /// This method is invoked immediately before each request is sent (including retries).
+    /// # Arguments
+    /// * `request` - Mutable reference to the HTTP request being sent
+    fn on_before_send_request(&self, request: &mut Request);
+
+    /// Determines whether to retry an operation that resulted in an error
+    ///
+    /// This method is called when an operation fails with an exception (network error,
+    /// timeout, or HTTP error). The implementation should examine the error and decide
+    /// whether the operation should be retried and how long to wait before retrying.
+    ///
+    /// # Arguments
+    /// * `err` - The error that occurred during the operation
+    ///
+    /// # Returns
+    /// `ShouldRetryResult` indicating:
+    /// - `should_retry`: Whether the operation should be retried
+    /// - `backoff_time`: How long to wait before retrying (if `should_retry` is true)
+    async fn should_retry_exception(&self, err: &azure_core::Error) -> ShouldRetryResult;
```

Comment:
Rust doesn't have exceptions, this should be `should_retry_error`. Honestly, we should just have a `should_retry` that takes `azure_core::Result<RawResponse>` and checks for Ok/Err itself.